test/test_nestedtensor.py

```diff
@@ -611,6 +679,9 @@ def test_dropout(self, device, dtype):
                     expect_tensor[j] /= 1.0 - p
         self.nt_equal(nt, expect)
 
+    # dropout works directly on the underlying buffer memory
+    # so contiguous / noncontiguous does not make any difference
```

Comment:
question: how to test this case? Since we redispatch to the buffer, we can only reproduce dropout result on 2 nested tensors with identical buffer.

I'm considering this approach:
1. define 2 nested tensors with identical buffer
2. reshape and transpose one of them
3. call dropout on both
4. revert the reshaping and transposing then compare

wdyt?

Reply:
oh, good point, let's not worry about this for now. In the future as we consider OpInfo integration, it may be nice to have some more automated way of testing non-contiguity. `ModuleInfo` has this now, for example - there's a test that systematically makes the sample inputs non-contiguous and ensures that outputs match those given by the reference.